program/src/error.rs

```diff
@@ -50,17 +50,34 @@ impl From<TokenWrapError> for ProgramError {
     }
 }
 
-impl<T> DecodeError<T> for TokenWrapError {
-    fn type_of() -> &'static str {
-        "TokenWrapError"
+impl TryFrom<u32> for TokenWrapError {
+    type Error = ();
```

Comment:
This is a bit of a strange error type. Would it be more useful to fall back to a `ProgramError`?